R/content.R

```diff
@@ -925,3 +975,55 @@ get_content_permissions <- function(content, add_owner = TRUE) {
   res <- content$permissions(add_owner = add_owner)
   parse_connectapi_typed(res, connectapi_ptypes$permissions)
 }
+
+#' Render a content item.
+#' 
+#' @description Submit a request to render a content item. Once submitted, the
+#' server runs an asynchronous process to render the content. This might be
+#' useful if content needs to be updated after its source data has changed,
+#' especially if this doesn't happen on a regular schedule.
+#' 
+#' Only valid for rendered content (e.g., most Quarto documents, Jupyter
+#' notebooks, R Markdown reports).
+#' 
+#' @return A [VariantTask] object that can be used to track completion of the render.
+#' @export
+content_render <- function(content) {
+  suppressWarnings({
+    validate_R6_class(content, "Content")
+    if (!content$is_rendered) {
+      stop(glue::glue("Render not supported for application mode: {content$content$app_mode}. Did you mean content_restart()?"))
+    }
+    render_task <- content$default_variant$render()
+    render_task$task_id <- render_task$id
+  
+    ContentTask$new(connect = content$get_connect(), content = content$get_content(), task = render_task)
+  })
+}
+
+#' Restart a content item.
+#' 
+#' @description Submit a request to restart a content item. Once submitted, the
+#' server performs an asynchronous request to kill all processes associated with
+#' the content item, starting new processes as needed. This might be useful if
+#' the application relies on data that is loaded at startup, or if its memory
+#' usage has grown over time.
+#' 
+#' Note that users interacting with certain types of applications may have their
+#' workflows interrupted.
+#' 
+#' Only valid for interactive content (e.g., applications, APIs).
+#' 
+#' @export
+content_restart <- function(content) {
+  validate_R6_class(content, "Content")
+  if (!content$is_interactive) {
+    stop(glue::glue("Restart not supported for application mode: {content$content$app_mode}. Did you mean content_render()?"))
+  }
+  random_hash = token_hex(32)
+  env_var_name = glue::glue("CONNECT_RESTART_{random_hash}")
```

Comment:
I used a leading `_` here to provide an extra hint that this variable is "private" by convention.

https://github.com/posit-dev/posit-sdk-py/blob/main/src/posit/connect/content.py#L217C21-L217C42

Reply:
Ok, I'll do the same. I'm not convinced that we _want_ to use a random hash for the var name here; I think it made something harder in testing, and the orphaned variable in an example I tried looked kinda weird in Connect's UI — I could imagine it making users worry that something is wrong in Connect's dashboard.

But I don't feel that way very strongly _at all_.

How long is the random strong the Python package uses?
```suggestion
  env_var_name = glue::glue("_CONNECT_RESTART_{random_hash}")
```

Reply:
32 characters. If the user ever sees it in the UI, then something has gone wrong.